app/helpers/components/filters_helper.rb

```diff
@@ -0,0 +1,43 @@
+module Components::FiltersHelper
+
+  def fe_filter_table_wrapper(url:, selector:, sorting:, ordering:, static_params: {}, &block)
```

Comment:
This is the functional component approach I am taling about. Much like rendering a shared/partial but we hand it off first inside a ruby function so that we can do complex operations/params processing BEFORE handing it off to the partial. This way we avoid bulky ruby logic inside .html.slim files while still retaining the ability to re-use components easily.

`fe_` prefix for these functions just to denote that these are `frontend` component functions.

Reply:
Moreover here, I personally prefer named parameters so that we technically require the necessary variables, much like setting locals in partials. Though in terms of locals, we can easily miss when local_assigns was not set correctly I think.

Reply:
We can add params **opts though, if we really need more customizations in the future if we have optional parameters that does not make sense adding as explicit function params, but for now this architecture should serve what our datatables need.